### PR TITLE
discrete protocol APIs

### PR DESCRIPTION
**Pull request by copejon**
Replaces a generic bucketIdentifier with API representations of each major object store protocol.

### CODE REVIEW

File: keps/sig-storage/20191125-bucket-provisioning.md [hunk 1]
```diff
@@ -255,12 +255,25 @@ spec:
   secretRef: [10]
     name:
     namespace:
-  bucketNameActual: [11]
-  bucketIdentifier: [12]
-  parameters: [13]
+  azureBlobIdentity: [11]
```
Comment (jeffvance) [inline keps/sig-storage/20191125-bucket-provisioning.md:258]: should the 3 store providers be under an umbrella stanza, eg. `bucketSource:` or `bucketContentIdentity:` ...?

File: keps/sig-storage/20191125-bucket-provisioning.md [hunk 2]
```diff
@@ -301,7 +316,7 @@ kind: BucketClass
 metadata:
   name: 
 provisioner: [1]
-supportedProtocols: [2]
+protocol: [2]
```
Comment (jeffvance) [inline keps/sig-storage/20191125-bucket-provisioning.md:319]: Do also want all or a subset of `BucketContentIdentity` in the `BucketClass`? Or will fields like _accountName_ be defined in _parameters_?

File: keps/sig-storage/20191125-bucket-provisioning.md (hunk 1 above)
Comment (copejon) [reply keps/sig-storage/20191125-bucket-provisioning.md:258]: for PVs, in code they are, but at the API they're embedded.  I figured we'd follow a similar pattern

File: keps/sig-storage/20191125-bucket-provisioning.md (hunk 2 above)
Comment (copejon) [reply keps/sig-storage/20191125-bucket-provisioning.md:319]: accountName i would think are returned by the driver.  we can't determine if a driver would provisioner new accounts per BucketContent, or a new key in a meta-account.

**Pull request merged**